functions to get routes from Bing API

Retr0Metal98 opened the pull request:
## Changes in this PR

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change of an existing feature
- [ ] Refactor/Performance enhancements

## Overview

- made functions to extract all the directed routes from our filled csv (`camera_dir_locs.csv`); for each route the info from Bing API is extracted and merged
- for now the csv filled by us should be on the local filesystem, and the result frame is output to local csv.

Retr0Metal98 committed:
functions to get routes from Bing API

made functions to extract all the directed routes from our filled csv (`camera_dir_locs.csv`)

for each route the info from Bing API is extracted and merged

for now the csv filled by us should be on the local filesystem, and the result frame is output to local csv.

Retr0Metal98 committed:
sample combined dataframe for congestion data

output from call to `process_camera_routes` function in `bing_map_ingest.py`

Retr0Metal98 committed:
Merge branch 'master' of https://github.com/don-tay/traffic-predict into bing-api-ingest

Retr0Metal98 committed:
bing data ingest: formatting corrections

Retr0Metal98 committed:
Merge branch 'master' of https://github.com/don-tay/traffic-predict into bing-api-ingest

Retr0Metal98 committed:
Update bing_map_ingest.py

added output options for bing congestion data, still need to decide where to pull the dir_locs csv file from

Retr0Metal98 committed:
Merge branch 'master' of https://github.com/don-tay/traffic-predict into bing-api-ingest

Retr0Metal98 committed:
separate function to produce route data to use

function `produce_route_table` converts:
sheet consisting of 1 row per camera
into
CSV with rows of camera, direction (multiple rows per camera)

sample of this route table in route_data.csv (actual route_data.csv should be pushed to S3 bucket).

`process_route_congestion` would takes the route table from S3 (for now generates route table in memory) and sequentially calls the Bing Routes API on each row (takes 50 sec for 155 rows). these calls could potentially be parallelized to speed up.

Retr0Metal98 committed:
complete congestion data ingestion

function process_route_congestion completed, will process ~180 routes per call.

added function to `ingest_all` in `task_schedule.py` for now, but should consider limit on number of API calls per 24 hr period. (can resolve with less frequent calls maybe?)

`camera_dir_locs.csv` has the finalized route start & end locations for each direction for each camera.

Retr0Metal98 committed:
function to merge bing csvs

Retr0Metal98 committed:
correction to camera_dir_locs.csv

don-tay reviewed (approved):
LGTM, for `camera_dir_locs.csv`, is this stored on S3? If so, we can add this file to the gitignore list

Retr0Metal98 committed:
limit bing api calls to specific time period

corrected the function to get timestamp based on SG time zone
perform collection of Bing data within a specific time-period to limit the number of calls per 24 hr.
